cts-java/src/test/java/org/ga4gh/cts/api/variants/VariantsSearchIT.java

```diff
@@ -0,0 +1,167 @@
+     */
+    @Test
+    public void checkExpectedVariantSets() throws AvroRemoteException {
+        final int expectedNumberOfVariantSets = 1;
+
+        final SearchVariantSetsRequest req =
+                SearchVariantSetsRequest.newBuilder()
+                                        .setDatasetId(TestData.getDatasetId())
+                                        .build();
+        final SearchVariantSetsResponse resp = client.variants.searchVariantSets(req);
+
+        final List<VariantSet> variantSets = resp.getVariantSets();
+        assertThat(variantSets).hasSize(expectedNumberOfVariantSets);
+    }
+
+    /**
+     * Check that the variants we receive from {@link org.ga4gh.ctk.transport.protocols.Client.Variants#searchVariants(SearchVariantsRequest)}
+     * search contain the expected reference name.
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @Test
+    public void checkVariantsForExpectedReferenceName() throws AvroRemoteException {
+        final long start = 50;
+        final long end = 100;
+
+        final SearchVariantSetsRequest req =
+                SearchVariantSetsRequest.newBuilder()
+                                        .setDatasetId(TestData.getDatasetId())
+                                        .build();
+        final SearchVariantSetsResponse resp = client.variants.searchVariantSets(req);
+
+        final List<VariantSet> variantSets = resp.getVariantSets();
+        assertThat(variantSets).isNotEmpty();
+        final String id = variantSets.get(0).getId();
+
+        final SearchVariantsRequest vReq =
+                SearchVariantsRequest.newBuilder()
+                                     .setVariantSetId(id)
+                                     .setReferenceName(TestData.REFERENCE_NAME)
+                                     .setStart(start).setEnd(end)
+                                     .build();
+        final SearchVariantsResponse vResp = client.variants.searchVariants(vReq);
+        final List<Variant> searchVariants = vResp.getVariants();
+
+        searchVariants.stream().forEach(v -> assertThat(v.getReferenceName()).isEqualTo(TestData.REFERENCE_NAME));
+
+        searchVariants.stream().forEach(v -> v.getCalls()
+                                              .stream()
+                                              .forEach(c -> assertThat(c).isNotNull()));
+    }
+
+    /**
+     * Check that the variants we receive from {@link org.ga4gh.ctk.transport.protocols.Client.Variants#searchVariants(SearchVariantsRequest)}
+     * search contain well-formed {@link Call}s.
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @Test
+    public void checkCallsForWellFormedness() throws AvroRemoteException {
+        final long start = 50;
+        final long end = 100;
+
+        final SearchVariantSetsRequest req =
```

Comment:
Although please don't create anything called helpers/utils; move
common code to someplace functionally logical.  Dumping ground
modules don't make it easy to understand code (e.g. the
common.avdl, which almost none of it is common to the API).

CH Albach <notifications@github.com> writes:
> This should probably be extracted to a helper / Utils, it's repeated in most of
> these test cases.